### PR TITLE
Enhancement: Disable Deployments

### LINKED ISSUE #14529: Add "disabled" property to deployments
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

3.x

### Describe the current behavior

Currently deployments have `paused` but this impacts only schedules and their associated flow runs.,

### Describe the proposed behavior

Add `disabled` property to deployment which disables all flow run creation mechanisms for that deployment.

- [ ] `disabled` property on Deployment
- [ ] New API routes for `/disable` and `/enable` 
- [ ] Calling `create_flow_run` returns an error
- [ ] Automation triggers that would create flow runs for the deployment instead do nothing
- [ ] When `/disable` is called, we delete the scheduled runs for the deployment
- [ ] The scheduling mechanism does not schedule new runs for `disabled` deployments
- [ ] Editing schedules for a `disabled` deployment returns an error

### Example Use

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dylanbhughes**
Closes #14529

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: ui/src/pages/Deployment.vue [hunk 1]
```diff
@@ -5,6 +5,7 @@
         :deployment="deployment"
         @update="deploymentSubscription.refresh"
         @delete="routeToDeployments"
+        :deployment-disable-toggle="true"
```
Comment (znicholasbrown) [inline ui/src/pages/Deployment.vue:8]:
```suggestion
        deployment-disable-toggle
```
The presence of this attr at all evaluates to `true`

File: ui/src/pages/Deployments.vue [hunk 2]
```diff
@@ -10,7 +10,7 @@
       </template>
 
       <template v-else>
-        <DeploymentList @delete="deploymentsSubscription.refresh" />
+        <DeploymentList @delete="deploymentsSubscription.refresh" :deployment-disable-toggle="true" />
```
Comment (znicholasbrown) [inline ui/src/pages/Deployments.vue:13]:
```suggestion
        <DeploymentList @delete="deploymentsSubscription.refresh" deployment-disable-toggle />
```
Same here

**Review by znicholasbrown — APPROVED** [review]
UI side of this looks good, just a minor nit

**Comment by dylanbhughes** [comment]
comment

**Comment by dylanbhughes** [comment]
comment

**Comment by dylanbhughes** [comment]
comment

**Comment by dylanbhughes** [comment]
Hey Team :wave::skin-tone-2: wanted to resurface this PR. I'm not 100% sure that [this issue](https://github.com/PrefectHQ/prefect/issues/14973) is really blocking. It doesn't seem like we intend to change the schedule reactivation behavior when a flow is deployed.

Since we're seeing some [user interest](https://github.com/PrefectHQ/prefect/issues/14529#issuecomment-2430161404) in the feature, I wanted to see if we could reignite discussion on the disabling deployments feature and see if there's a path to getting this merged.

cc @aaazzam @zzstoatzz @cicdw

**Comment by desertaxle** [comment]
@dylanbhughes I think the issue that @aaazzam called out is tangentially related since changing the disabled/enabled state of a deployment via any of our SDK methods could change the schedule state unexpectedly. That's an issue with the SDK and not this work, so I think we could find a way to implement this without first resolving that issue.

Here's a potential way to add this feature while steering clear of our current footguns: this feature starts as an API/UI feature only. This will allow users to disable deployments from the UI where it's most likely to be used, and we can add this feature to the SDK once we've improved our deployment update story.

Happy to hear pushback or other ideas!

**Comment by dylanbhughes** [comment]
I think that sounds great! Disabling or enabling a deployment feels like it should be a pretty explicit action and probably shouldn't occur as a side effect of another action anyway. I'll clean this PR up and do some testing -- hoping to have something by EOW/early next week.

**Comment by dylanbhughes** [comment]
comment

**Comment by dylanbhughes** [comment]
Comment

**Pull request closed without merging**